addons/l10n_in/models/account_tax.py

```diff
@@ -0,0 +1,118 @@
+from odoo import fields, models, api
+from odoo.tools import frozendict
+
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    l10n_in_reverse_charge = fields.Boolean("Reverse charge", help="Tick this if this tax is reverse charge. Only for Indian accounting")
+
+    def _prepare_dict_for_taxes_computation(self):
+        # EXTENDS 'account'
+        tax_values = super()._prepare_dict_for_taxes_computation()
+
+        if self.country_code == 'IN':
+            l10n_in_tax_type = None
+            tags = self.invoice_repartition_line_ids.tag_ids
+            if self.env.ref('l10n_in.tax_tag_igst') in tags:
+                l10n_in_tax_type = 'igst'
+            elif self.env.ref('l10n_in.tax_tag_cgst') in tags:
+                l10n_in_tax_type = 'cgst'
+            elif self.env.ref('l10n_in.tax_tag_sgst') in tags:
+                l10n_in_tax_type = 'sgst'
+            elif self.env.ref('l10n_in.tax_tag_cess') in tags:
+                l10n_in_tax_type = 'cess'
+            tax_values['_l10n_in_tax_type'] = l10n_in_tax_type
+
+        return tax_values
+
+    # -------------------------------------------------------------------------
+    # HELPERS IN BOTH PYTHON/JAVASCRIPT (hsn_summary.js / account_tax.py)
+
+    # HSN SUMMARY
+    # -------------------------------------------------------------------------
+
+    @api.model
+    def _l10n_in_get_hsn_summary_table(self, base_lines, display_uom):
+        results_map = {}
+        l10n_in_tax_types = set()
+        for base_line in base_lines:
+            l10n_in_hsn_code = base_line['l10n_in_hsn_code']
+            if not l10n_in_hsn_code:
+                continue
+
+            price_unit = base_line['price_unit']
+            quantity = base_line['quantity']
+            uom = base_line['uom'] or {}
+            tax_values_list = base_line['tax_values_list']
+
+            # Compute the taxes.
+            eval_context = self.env['account.tax']._eval_taxes_computation_prepare_context(
```

Comment:
This has nothing to do with (`safe_`)`eval`, right?
Then change the name to avoid pinging security :wink: 
```suggestion
            evaluation_context = self.env['account.tax']._eval_taxes_computation_prepare_context(
```